.github/workflows/buildAndTest.yml

```diff
@@ -1,182 +1,193 @@
+      if: matrix.llvm-build == 'in-tree'
       run: |
-        cd $GITHUB_WORKSPACE
         export PYTHONPATH="$GITHUB_WORKSPACE/build/tools/torch-mlir/python_packages/torch_mlir"
         python -m e2e_testing.torchscript.main --config=eager_mode -v
 
-    - name: Run TOSA backend - TorchScript end-to-end tests
-      if: matrix.llvmtype == 'binary'
+    - name: Run tosa e2e integration tests
+      if: matrix.llvm-build == 'in-tree'
       run: |
-        cd $GITHUB_WORKSPACE
         export PYTHONPATH="$GITHUB_WORKSPACE/build/tools/torch-mlir/python_packages/torch_mlir"
         python -m e2e_testing.torchscript.main --config=tosa -v
 
-    - name: Run Lazy Tensor Core - TorchScript end-to-end tests
-      if: matrix.llvmtype == 'binary'
+    - name: Run lazy_tensor_core e2e integration tests
+      if: matrix.llvm-build == 'in-tree'
       run: |
-        cd $GITHUB_WORKSPACE
         export PYTHONPATH="$GITHUB_WORKSPACE/build/tools/torch-mlir/python_packages/torch_mlir"
         python -m e2e_testing.torchscript.main --config=lazy_tensor_core -v
+
+
+  macos-build:
+    name: macos-arm64
+    runs-on: macos-12
+    strategy:
+      fail-fast: true
+      matrix:
+        llvm-build: [in-tree, out-of-tree]
+        torch-binary: [ON, OFF]
+        exclude:
+          # Exclude llvm in-tree and pytorch source
+          - llvm-build: in-tree
+            torch-binary: OFF
+          # Exclude llvm out-of-tree altogether
+          - llvm-build: out-of-tree
+
+    steps:
+    - name: Checkout torch-mlir
+      uses: actions/checkout@v2
+      with:
+        submodules: 'true'
+
+    - name: Setup ccache
+      uses: ./.github/actions/setup-build
+      with:
+        cache-suffix: macos-arm64-${{ matrix.llvm-build }}-${{ matrix.torch-binary }}
+
+    - name: Configure llvm-build='in-tree' torch-binary='${{ matrix.torch-binary}}'
+      # libzstd on GH Runners are only x86_64 to remove them.
+      run: |
+        cmake -GNinja -Bbuild_arm64 \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_C_COMPILER=clang \
+          -DCMAKE_CXX_COMPILER=clang++ \
+          -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+          -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+          -DCMAKE_LINKER=lld \
+          -DCMAKE_OSX_ARCHITECTURES=arm64 \
+          -DLLVM_ENABLE_ASSERTIONS=ON \
+          -DLLVM_ENABLE_PROJECTS=mlir \
+          -DLLVM_EXTERNAL_PROJECTS="torch-mlir;torch-mlir-dialects" \
+          -DLLVM_EXTERNAL_TORCH_MLIR_SOURCE_DIR="$GITHUB_WORKSPACE" \
+          -DLLVM_EXTERNAL_TORCH_MLIR_DIALECTS_SOURCE_DIR="${GITHUB_WORKSPACE}/external/llvm-external-projects/torch-mlir-dialects" \
+          -DLLVM_TARGETS_TO_BUILD=AArch64 \
+          -DLLVM_USE_HOST_TOOLS=ON \
+          -DLLVM_ENABLE_ZSTD=OFF \
+          -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
+          -DTORCH_MLIR_ENABLE_LTC=OFF \
+          -DTORCH_MLIR_ENABLE_MHLO=OFF \
+          -DTORCH_MLIR_USE_INSTALLED_PYTORCH="${{ matrix.torch-binary }}" \
```

Comment:
We can delete this https://github.com/llvm/torch-mlir/blob/51bfe25c898e5f7a09294ccc012cfbf1327614c9/build_tools/build_libtorch.sh#L134 in either this commit or a follow on and switch to src.